core/templates/services/math-interactions.service.spec.ts

```diff
@@ -27,82 +27,256 @@ describe('MathInteractionsService', () => {
+      'Please enter an equation instead.');
+
+    expect(mathInteractionsService.validateEquation(
+      '5 >= 2*alpha')).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'It looks like you have entered an inequality. ' +
+      'Please enter an equation instead.');
+
+    expect(mathInteractionsService.validateEquation(
+      '2*x^2 + 3')).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'It looks like you have entered an expression. ' +
+      'Please enter an equation instead.');
+
+    expect(mathInteractionsService.validateEquation('(x+y)/0 = 5')).toBeFalse();
     expect(mathInteractionsService.getWarningText()).toBe(
       'Division by zero not allowed.');
 
-    expect(mathInteractionsService.validateAnswer('(x+y)/(y-y)')).toBeFalse();
+    expect(mathInteractionsService.validateEquation(
+      '3*x^2 = (x+y)/(y-y)')).toBeFalse();
     expect(mathInteractionsService.getWarningText()).toBe(
       'Division by zero not allowed.');
 
-    expect(mathInteractionsService.validateAnswer('a)(b')).toBeFalse();
+    expect(mathInteractionsService.validateEquation('a)(b = x')).toBeFalse();
     expect(mathInteractionsService.getWarningText()).toBe(
       'It looks like your answer has an invalid bracket pairing.');
 
-    expect(mathInteractionsService.validateAnswer('3.4.5 + 45/a')).toBeFalse();
+    expect(mathInteractionsService.validateEquation(
+      '3.4.5 = 45/a')).toBeFalse();
     expect(mathInteractionsService.getWarningText()).toBe(
       'Invalid integer.');
   });
+
+  it('should insert missing multiplication signs', function() {
+    expect(mathInteractionsService.insertMultiplicationSigns(
+      'ab/2')).toBe('a*b/2');
+    expect(mathInteractionsService.insertMultiplicationSigns(
+      'alpha+ax^2')).toBe('alpha+a*x^2');
+    expect(mathInteractionsService.insertMultiplicationSigns(
+      'sqrt(xyz)')).toBe('sqrt(x*y*z)');
+    expect(mathInteractionsService.insertMultiplicationSigns(
+      'ax^2+2*ab+b^2')).toBe('a*x^2+2*a*b+b^2');
+    expect(mathInteractionsService.insertMultiplicationSigns(
+      'cos(theta/ab)+sin(xy)')).toBe('cos(theta/a*b)+sin(x*y)');
+    expect(mathInteractionsService.insertMultiplicationSigns(
+      'log(alpha/pi)')).toBe('log(alpha/pi)');
+    expect(mathInteractionsService.insertMultiplicationSigns(
+      'Al^2')).toBe('A*l^2');
+  });
+
+  it('should replace abs symbol with text', function() {
+    expect(mathInteractionsService.replaceAbsSymbolWithText(
+      '|x|')).toBe('abs(x)');
+    expect(mathInteractionsService.replaceAbsSymbolWithText(
+      '40alpha/|beta|')).toBe('40alpha/abs(beta)');
+    expect(mathInteractionsService.replaceAbsSymbolWithText(
+      'abs(xyz)')).toBe('abs(xyz)');
+    expect(mathInteractionsService.replaceAbsSymbolWithText(
+      '|sqrt(a+b^2)|')).toBe('abs(sqrt(a+b^2))');
+    expect(mathInteractionsService.replaceAbsSymbolWithText(
+      '||')).toBe('abs()');
```

Comment:
shouldn't this be invalid?

Reply:
Yes, it will be categorized as invalid. This is just to test that the replacement function works as expected. When the validation will be called on this expression it will evaluate it as invalid.

Reply:
Sounds good!